superset/viz.py

```diff
@@ -225,44 +223,41 @@ def get_json(self, force=False):
             self.get_payload(force),
             default=utils.json_int_dttm_ser, ignore_nan=True)
 
-    @property
-    def cache_key(self):
-        form_data = self.form_data.copy()
-        merge_extra_filters(form_data)
-        s = str([(k, form_data[k]) for k in sorted(form_data.keys())])
+    def cache_key(self, query_obj):
+        s = self.datasource.get_query_str(query_obj)
```

Comment:
This makes it such that `get_query_str` needs to be deterministic. I think that's the case. Let's note that if we see cache-hit-misses in the future it may be related to `get_query_str` not being fully deterministic.

Reply:
Correct. I'll add a docstring to provide more context.